### ISSUE TITLE
Rollout policy: the rollouts are NOT greedy today (temp 0.5) — greedy vs eps-greedy vs softmax (09/03 action item 4)

### ISSUE THREAD

**Issue by manchado53** [manchado53/counterfactual-reasoning#20]
From the 09/03 touchpoint with Jeremy (action item 4). Notion: *RL Research Meeting:
Replay Buffer Analysis and CCE Scoring*, 2026-09-03.

> Try using an epsilon-greedy exploration policy during rollouts instead of the greedy
> policy to reduce compute.

Jeremy's compute argument, which is correct: under epsilon-greedy you only need the
Q-values on exploit steps, so you can skip the forward pass on the explore steps.

## First: the premise about the current code is wrong

The discussion assumed the rollouts currently use a **greedy** policy. They do not.
`agents/jax_nav/config.py:75` sets `cf_rollout_temperature: 0.5`, and
`consequence_dqn._build_rollout_fn` branches on it:

```python
if temperature and temperature > 0:
    action = jax.random.categorical(ak, q / temperature)   # <-- THIS path runs
else:
    action = jnp.argmax(q)
```

Every JaxNav sweep to date (bootstrap 274476, instrumentation 274678) ran
`cf_rollout_temperature=0.5`, i.e. a **softmax at temperature 0.5** — already stochastic,
and already off-policy relative to the greedy policy being learned. So:

- Jeremy's recommendation to *use the greedy policy* would be a **change** from current
  behaviour, not a confirmation of it.
- The compute saving from epsilon-greedy does **not** apply as stated: a softmax needs the
  full Q-vector on every step, so there is no forward pass to skip today. Epsilon-greedy
  would be the thing that *creates* the saving.
- The "in a deterministic environment a single rollout may suffice" reasoning also does
  not describe the current runs — the temperature is exactly what spreads the N rollouts
  in a deterministic environment, which is what the code comment says it is for.

## Second: this is measurably distorting the score

Instrumented 2026-09-03 (`fig_instrumentation.png` panel 3): rollout terminal reasons
against real eval outcomes, `cce_wmean`, array 274678.

```
                imagined (temp 0.5)    real (argmax)
   crash              57%                   11%
   goal               18%                   45%
```

The rollout policy crashes ~5x more and reaches the goal ~2.5x less than the policy
actually being trained. The two curves agree while the agent is still bad and diverge as
it becomes competent — so **the error grows as the agent improves.** CCE is scoring
counterfactual returns under a policy the agent does not have.

(Measured to 178k of 250k episodes. Confirm on the finished run before quoting.)

## Suggested experiment

Three arms, `cf_rollout_temperature` as the only difference, on `cce_wmean`, 8x8_f03:

```
   0.5   current shipped behaviour (softmax)
   0.0   greedy -- Jeremy's recommendation, and the learned policy in DQN
   eps   epsilon-greedy at the training epsilon, the compute-saving variant
```

Report: rollout-vs-eval outcome gap (the table above), the zero-rate of measurements,
`scored_frac`, and wall clock. The greedy arm should close the outcome gap; the question
is whether it also collapses the score to {0,1} the way FL-det does (issue #13), which
would trade one problem for another.

Note the epsilon-greedy compute saving needs an implementation change, not just a config
flag — the current branch computes `q = network.apply(params, obs)` unconditionally
before choosing.
